Index: personal-dwm/dwm/Dockerfile
===================================================================
--- /dev/null
+++ personal-dwm/dwm/Dockerfile
@@ -0,0 +1,4 @@
+FROM debian:stable
+RUN echo "deb-src http://deb.debian.org/debian stable main" >> /etc/apt/sources.list
+RUN apt-get update -q
+RUN apt-get build-dep -y dwm
Index: personal-dwm/dwm/Makefile
===================================================================
--- personal-dwm.orig/dwm/Makefile
+++ personal-dwm/dwm/Makefile
@@ -57,4 +57,8 @@ uninstall:
 	@echo removing manual page from ${DESTDIR}${MANPREFIX}/man1
 	@rm -f ${DESTDIR}${MANPREFIX}/man1/dwm.1
 
+docker-build:
+	@docker build -t dwm-builder .
+	@docker run --rm -v "$(shell pwd):/work" -w /work -u "$(shell id -u)" dwm-builder make all
+
 .PHONY: all options clean dist install uninstall
